cloudweatherreport/run.py

```diff
@@ -1,33 +1,231 @@
 from __future__ import print_function
 
-import subprocess
-import sys
+import multiprocessing as mp
+import argparse
+from cStringIO import StringIO
+from datetime import datetime
+import logging
+import os
+import traceback
+from copy import copy
 
-from cloud_weather_report import (
-    main,
-    parse_args
+from bundletester import tester
+
+from cloudweatherreport import model
+from cloudweatherreport.datastore import DataStore
+from cloudweatherreport.utils import (
+    configure_logging,
+    connect_juju_client,
+    find_unit,
+    fetch_svg,
+    get_provider_name,
+    run_action,
+    get_bundle_yaml,
+    get_juju_major_version,
+    get_versioned_juju_api,
+    generate_test_id,
 )
-from utils import read_file
 
 
-def file_open_with_app(filename):
-    opener = {'linux2': 'xdg-open', 'linux': 'xdg-open', 'darwin': 'open',
-              'win32': 'start'}
-    try:
-        subprocess.call([opener[sys.platform], filename])
-    except:
-        pass
+def parse_args(argv=None):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('controllers', nargs='+', help="Controller list.")
+    parser.add_argument('test_plan', help="Test plan YAML file.")
+    parser.add_argument('--results-dir', default='results',
+                        help="Directory to store / find results.")
+    parser.add_argument('--bucket',
+                        help='Store / find results in this S3 bucket '
+                             'instead of locally')
```

Comment:
This allows us to store data at root level only. I think we should also have another option for `prefix` folder.  With the option provided we can only store at `/*`.

Currently, cwr test results are stored in the `/juju-qa-data/cwr/*` folder.